lib/analytics.js

```diff
@@ -14,16 +14,32 @@ if (process.env.REDISTOGO_URL) {
 }
 
 let analytics = {};
+let autosaveIntervalId;
 
 const analyticsAutoSaveFileName = process.env.SHIELDS_ANALYTICS_FILE || './analytics.json';
-const analyticsAutoSavePeriod = 10000;
-setInterval(function analyticsAutoSave() {
+
+function performAutosave() {
+  const contents = JSON.stringify(analytics);
   if (useRedis) {
```

Comment:
I feel like this should use the adapter pattern rather than hard-coding Redis vs plain text file. That could definitely be refactored separately though!

Reply:
Yea, agreed. We're not even using Redis in production so I don't know if this code works. We could remove it, though it would make more sense to get some tests around it, since it makes our hosting portable to Heroku for people who want to self-host. Definitely agreed it would be nice to refactor it.